### PR TITLE
Fix distro and instrumentation version values

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/signalfx/splunk-otel-go/issues/1988

- Fix `telemetry.sdk.version` resource attribute to properly return the `github.com/signalfx/splunk-otel-go/distro` Go module version.
- Fix instrumentation scope version value to properly return the instrumentation library versions.

### CODE REVIEW

**Review by theletterf — APPROVED** [review]

File: instrumentation/database/sql/splunksql/test/suite_test.go [hunk 1]
```diff
@@ -307,6 +308,7 @@ func (s *SplunkSQLSuite) assertSpans(name moniker.Span, count int, c traceapi.Sp
 			}
 		}
 		s.Equalf(traceapi.SpanKindClient, roSpan.SpanKind(), "span %q is not a client span", name)
+		s.Equal(splunkotel.Version(), roSpan.InstrumentationLibrary().Version, "version should match")
```
Comment (MrAlias) [inline instrumentation/database/sql/splunksql/test/suite_test.go:311]: Shouldn't this test against the version from `instrumentation/database/sql/splunksql`?

And similar for the other `test` packages?

Comment (MrAlias) [reply instrumentation/database/sql/splunksql/test/suite_test.go:311]: I think we can just export `Version`. That is what we do upstream when we need to reference the version like this.

I would refrain from adding a version to the test packages as they are only intended to test their respective instrumentation.

Comment (pellared) [reply instrumentation/database/sql/splunksql/test/suite_test.go:311]: How about addressing it in a separate PR?  I can create an issue for tracking. I plan to solve it on Tuesday.

Comment (pellared) [reply instrumentation/database/sql/splunksql/test/suite_test.go:311]: The good thing about doing it in a separate PR is that this PR will be a fix and the next one would expand the API.

Comment (pellared) [reply instrumentation/database/sql/splunksql/test/suite_test.go:311]: https://github.com/signalfx/splunk-otel-go/issues/1992

**Review by MrAlias — APPROVED** [review]

**Pull request merged**